paasta_tools/chronos_tools.py

```diff
@@ -420,12 +424,9 @@ def format_chronos_job_dict(self, docker_url, docker_volumes, dockercfg_location
         if self.get_schedule() is not None:
             complete_config['schedule'] = self.get_schedule()
         else:
-            matching_parent_pairs = [(parent, get_job_for_service_instance(*parent.split(".")))
-                                     for parent in self.get_parents()]
-            for parent_pair in matching_parent_pairs:
-                if parent_pair[1] is None:
-                    raise InvalidParentError("%s has no matching jobs in Chronos" % parent_pair[0])
-            complete_config['parents'] = [parent_pair[1]['name'] for parent_pair in matching_parent_pairs]
+            # The input to parents is the normal paasta syntax, but for chronos we have to
+            # convert it to what chronos expects, which uses its own spacer
+            complete_config["parents"] = map(paasta_to_chronos_job_name, self.get_parents())
```

Comment:
the reason we did this pre-fetching was so that we could skip setting up a job if it's parents didn't yet exist in ``setup_chronos_job``. [here's the code where we caught this](https://github.com/Yelp/paasta/blob/master/paasta_tools/setup_chronos_job.py#L208-L210). without the validation, we fail later (when the request to Chronos fails) and less gracefully (we don't catch the right exception). this is the reason itests are failing atm. I don't mind failing later, but I think it's atleast worth catching this specific error in setup_chronos_job if it fails.

also, I'm surprised flake let you get away with the ``map`` - can you make it a list comprehension instead?